### PR TITLE
Exchange addresses which cannot be used as an output of private transaction

### PR DESCRIPTION
**Pull request by psolstice**
## PR intention
As a requirement from some exchanges we introduce a new type of transparent address for which it's not possible to use it as an output for both private and coinbase transactions

## Code changes brief
A new opcode is introduced and new address prefix. The rest of the structure of private address and transactions is left intact. It is recommended to use `getnewexchangeaddress` to create a pair for existing transparent address. Exchanges publish new address as a mean to deposit funds. Normal transparent address can used as a source for sending the funds further.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -209,6 +209,59 @@ UniValue getnewaddress(const JSONRPCRequest& request)
...
 
+UniValue getnewexchangeaddress(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() > 1)
+        throw std::runtime_error("");
```
Comment (levonpetrosyan93) [inline src/wallet/rpcwallet.cpp:220]: why we don't throw  JSONRPCError here too?

**Review by levonpetrosyan93 — DISMISSED** [review]

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (psolstice) [reply src/wallet/rpcwallet.cpp:220]: this particular RPC it's not supposed to be documented at all

File: src/validation.cpp [hunk 2]
```diff
@@ -655,16 +672,22 @@ bool CheckTransaction(const CTransaction &tx, CValidationState &state, bool fChe
...
 
         if (tx.IsLelantusTransaction()) {
+            if (hasExchangeUTXOs)
+                return state.DoS(100, false, REJECT_INVALID, "bad-super-transparent");
             if (!CheckLelantusTransaction(tx, state, hashTx, isVerifyDB, nHeight, isCheckWallet, fStatefulZerocoinCheck, sigmaTxInfo, lelantusTxInfo))
                 return false;
         }
 
         if (tx.IsSparkTransaction()) {
+            if (hasExchangeUTXOs)
```
[the remark this answers is not in the document]
Comment (psolstice) [reply src/validation.cpp:696]: Did this in the latest commit

[the remark this answers is not in the document]
Comment (levonpetrosyan93) [reply src/validation.cpp:696]: It looks good.

**Review by levonpetrosyan93 — APPROVED** [review]

**Review by levonpetrosyan93 — APPROVED** [review]

**Pull request merged**